articles/introduction-to-recursion/index.md

```diff
@@ -0,0 +1,87 @@
+
+### Fibonacci Numbers
+We are going to write a program to return the "n"th Fibonacci number. For example, the 5th Fibonacci number is 5, and the 7th is 13. Our function will take one argument: n(the term we wish to calculate)
+
+```py
+def fibonacci(n):
+
+	if(n == 1 or n == 2):
+
+		return 1
+	  
+	return fibonacci(n - 1) + fibonacci(n - 2)
+  
+n = int(input('Enter the nth term: '))
+
+print(fibonacci(n))
+```
+
+The `if(n == 1 or n == 2)` condition in the function is called a "terminating condition" and is essential for every recursive function, as a function without this condition will keep calling itself and never terminate(infinite loop causes the program to crash). We know that the first two terms are 1, so we can return 1 when n is 1 or 2. This is also called a "base case" or a "base condition", and is usually something that we already know. For instance, we might know the answer for n = 0 or some other specific value of n. If we know this "base" value, we can use it to calculate all the other values.
+ 
+### Factorial
+A factorial is a mathematical expression that most students would have come across. The factorial of a number is defined as the product of all positive integers less than or equal to that number. In other words, if `n` is a positive integer, the factorial of n will be:  
+
+`factorial(n) = n * (n - 1) * (n - 2) * ........... * (3) * (2) * (1)`
+  
+For example,
+
+`factorial(4) = 4*3*2*1 = 24`
+
+`factorial(5) = 5*4*3*2*1 = 120`
+
+`factorial(6) = 6*5*4*3*2*1 = 720`
+
+`factorial(1) = 1*1 = 1`
+
+One interesting pattern that we observe here is that the factorial of 'n' is equal to `n` multiplied by `factorial(n - 1)`. In other words, `factorial(5) = 5*factorial(4)`. Similarly, `factorial(6) = 6*factorial(5)`. In general, `factorial(n) = n*factorial(n - 1)`. Interestingly, this looks exactly like a recursive function in which we call the 'factorial' function multiple times with different arguments. We know that `factorial(1) = 1`, which will be the terminating condition for our function.
+
+`Note: An important rule in mathematics is that the factorial of 0 is 1. In other words, factorial(0) = 1. We are going to use this as the base case of our function.`
+
+Code:
+
+```py
+def factorial(n):
+
+	if(n == 0 or n == 1):
+
+		return 1
+
+	return n*factorial(n - 1)
+
+n = int(input('Enter the number: '))
+
+print(factorial(n))
+```
+
+In the above function, the terminating condition, as stated above, is: `if(n == 0 or n == 1)`, we return 1 since `factorial(1) = factorial(0) = 1`. Otherwise, we use the formula described above to calculate the factorial when n is greater than 1.
```

Comment:
```suggestion
In the above function, the terminating condition, as stated above, is: `(n == 0 or n == 1)`. We return 1 since `factorial(1) = factorial(0) = 1`. Otherwise, we use the formula described above to calculate the factorial when n is greater than 1.
```